### PR TITLE
Move proctored exams to edx

### LINKED ISSUE #4636: Exams: update exam card for edx.org
#### Current Text:

Final Proctored Exam

You must take a proctored exam for each course. Exams may be taken at any authorized Pearson test center. Before you can take an exam, you have to pay for the course and pass the online work.

[gray block with profile information]

#### New Text:

Final Proctored Exam

You must take an online proctored exam for each course. Before you can take an exam, you have to pay for the course and pass the online work.

Exams will be available online on edX.org. You may take the exam at any time during the exam period. No advance scheduling is required, but you should verify your account and complete the exam onboarding as soon as possible. 

You are authorized to take the following exams, before [date] [time] [time zone]"

•  list of links to each approved exam 

## Acceptance Criteria:

- [ ] Update the text
- [ ] Remove the block about profiles and addresses. These are no longer necessary
- [ ] provide a link for each course that links to the appropriate course on edX and includes an embedded enrollment code

#### Out of Scope

- update course cards

### PR DESCRIPTION
**Pull request by annagav**
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
  - [ ] Tag @ferdi or @pdpinch for review
- [ ] Migrations
  - [ ] Migration is backwards-compatible with current production code
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested
- [ ] Settings
  - [ ] New settings are documented and present in `app.json`
  - [ ] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
Fix #4636

#### What's this PR do?
Update the exam card and the course row with information about taking exams on edx.

#### How should this be manually tested?
set `FEATURE_ENABLE_EDX_EXAMS` to true in .env file

Screen Shot 2020-09-04 at 9 14 35 AM

Screen Shot 2020-09-04 at 11 03 46 AM

### CODE REVIEW

**Comment by annagav** [comment]
Screen Shot 2020-09-10 at 2 58 54 PM

**Comment by pdpinch** [comment]
I think we'll need to change this message too: 

<image src="https://user-images.githubusercontent.com/7574259/90434088-db79cf00-e09a-11ea-8d26-7d8657194077.png">

We should try to re-use as much of the phrasing as possible, i.e. "You did not pass the exam. You are authorized to take the virtual proctored exam again. If you wish to take the proctored exam this testing session, you must complete the exam onboarding." (with "complete the exam onboarding" linked to edx.org)

File: static/js/components/dashboard/FinalExamCard.js [hunk 1]
```diff
@@ -267,28 +267,55 @@ export default class FinalExamCard extends React.Component<void, Props, void> {
...
+              </div>
+              <div className="exam-text">
+                <h2>Final Proctored Exam</h2>
+                <p>
+                  To earn a certificate, you must take an online proctored exam
+                  for each course. Before you can take a proctored exam, you
+                  have to pay for the course and pass the online work.
+                </p>
+                <p>
+                  Exams will be available online on edX.org. You may take the
```
Comment (noisecapella) [inline static/js/components/dashboard/FinalExamCard.js:309]: Should `edX.org` be a link?

File: dashboard/utils.py [hunk 2]
```diff
@@ -446,6 +447,16 @@ def get_pearson_exam_status(self):  # pylint: disable=too-many-return-statements
         except ExamProfile.DoesNotExist:
             return ExamProfile.PROFILE_ABSENT
 
+        if settings.FEATURES.get('ENABLE_EDX_EXAMS', False):
+            auths = ExamAuthorization.objects.filter(
+                user=user,
+                status=ExamAuthorization.STATUS_SUCCESS
+            )
+            if auths.exists():
+                return ExamProfile.PROFILE_SCHEDULABLE
```
Comment (noisecapella) [inline dashboard/utils.py:456]: Is this related to your comment on the issue? https://github.com/mitodl/micromasters/issues/4636#issuecomment-680239395

File: static/js/components/dashboard/FinalExamCard.js [hunk 3]
```diff
@@ -267,28 +267,55 @@ export default class FinalExamCard extends React.Component<void, Props, void> {
...
-      return schedulableCard(
-        profile,
-        program,
-        navigateToProfile,
-        pearson,
-        showPearsonTOSDialog,
-        pearsonTOSDialogVisible,
-        submitPearsonSSO
+    if (!SETTINGS.FEATURES.ENABLE_EDX_EXAMS) {
+      switch (program.pearson_exam_status) {
```
Comment (noisecapella) [inline static/js/components/dashboard/FinalExamCard.js:271]: Is this part deprecated now?

File: dashboard/utils.py [hunk 4]
```diff
@@ -446,6 +447,16 @@ def get_pearson_exam_status(self):  # pylint: disable=too-many-return-statements
         except ExamProfile.DoesNotExist:
             return ExamProfile.PROFILE_ABSENT
 
+        if settings.FEATURES.get('ENABLE_EDX_EXAMS', False):
+            auths = ExamAuthorization.objects.filter(
```
Comment (noisecapella) [inline dashboard/utils.py:451]: Can you add a test for this part?

File: static/js/components/dashboard/courses/StatusMessages.js [hunk 5]
```diff
@@ -113,6 +113,47 @@ const messageForAttemptedExams = (course: Course, passedExam: boolean) => {
...
+          "You are authorized to take the virtual proctored exam for this course. Please "
+        }
+        <a href="http://edx.org">
+          enroll now and complete the exam onboarding.
+        </a>
+      </span>
+    )
+  } else if (!R.isEmpty(course.exams_schedulable_in_future)) {
+    message =
+      "You can take the exam starting on " +
```
Comment (noisecapella) [inline static/js/components/dashboard/courses/StatusMessages.js:133]: Can you add or adjust a test to cover this condition?

File: static/js/components/dashboard/courses/StatusMessages.js [hunk 6]
```diff
@@ -113,6 +113,47 @@ const messageForAttemptedExams = (course: Course, passedExam: boolean) => {
...
+      `on ${formatDate(course.exams_schedulable_in_future[0])}.`
+  }
+  return message
+}
+
+const messageForAttemptedEdxExams = (course: Course, passedExam: boolean) => {
+  const whenFailed = passedExam ? "" : "You did not pass the exam. "
+  if (course.has_to_pay) {
+    return `${whenFailed}If you want to re-take the exam, you need to pay again.`
```
Comment (noisecapella) [inline static/js/components/dashboard/courses/StatusMessages.js:142]: Can you add a test for this condition?

File: static/js/components/dashboard/courses/StatusMessages.js [hunk 7]
```diff
@@ -113,6 +113,47 @@ const messageForAttemptedExams = (course: Course, passedExam: boolean) => {
...
+    return (
+      <span>
+        {`${whenFailed}You are authorized to take the virtual proctored exam for this course. Please `}
+        <a href="http://edx.org">
+          enroll now and complete the exam onboarding.
+        </a>
+      </span>
+    )
+  }
+  return `${whenFailed}`
```
Comment (noisecapella) [inline static/js/components/dashboard/courses/StatusMessages.js:154]: Can you add a test for this condition, when neither of the above are true?

File: static/js/components/dashboard/courses/StatusMessages.js [hunk 8]
```diff
@@ -311,12 +352,23 @@ export const calculateMessages = (props: CalculateMessagesProps) => {
...
-        ? messageForAttemptedExams(course, passedExam)
-        : messageForNotAttemptedExam(course)
+
+    if (SETTINGS.FEATURES.ENABLE_EDX_EXAMS) {
+      if (!passedExam) {
+        message = failedExam
+          ? messageForAttemptedEdxExams(course, passedExam)
+          : messageForNotAttemptedEdxExam(course)
+      } else {
+        message = messageForAttemptedEdxExams(course, passedExam)
```
Comment (noisecapella) [inline static/js/components/dashboard/courses/StatusMessages.js:362]: Can you add a test for this case?

**Review by noisecapella — CHANGES_REQUESTED** [review]

File: static/js/components/dashboard/FinalExamCard.js (hunk 1 above)
Comment (annagav) [reply static/js/components/dashboard/FinalExamCard.js:309]: no the link will come from a a csv file. This work hasn't been done yet.

File: dashboard/utils.py (hunk 2 above)
Comment (annagav) [reply dashboard/utils.py:456]: The exam profiles are loosing their meaning because we don't need Pearson to verify them anymore. So this is cutting down on other Profile status options.

File: static/js/components/dashboard/FinalExamCard.js (hunk 3 above)
Comment (annagav) [reply static/js/components/dashboard/FinalExamCard.js:271]: This value is still used to determine if the state of the exam card. It needs to be renamed, but I guess I thought to postpone it. Do you think it is better to rename it in this PR?

File: static/js/components/dashboard/courses/StatusMessages.js [hunk 9]
```diff
@@ -113,6 +113,47 @@ const messageForAttemptedExams = (course: Course, passedExam: boolean) => {
...
+  let message =
+    "There are currently no exams available. Please check back later."
+
+  if (course.can_schedule_exam) {
+    message = (
+      <span>
+        {
+          "You are authorized to take the virtual proctored exam for this course. Please "
+        }
+        <a href="http://edx.org">
```
Comment (noisecapella) [inline static/js/components/dashboard/courses/StatusMessages.js:126]: Can you change this to `https://edx.org`?

File: static/js/components/dashboard/courses/StatusMessages.js [hunk 10]
```diff
@@ -113,6 +113,47 @@ const messageForAttemptedExams = (course: Course, passedExam: boolean) => {
...
+const messageForAttemptedEdxExams = (course: Course, passedExam: boolean) => {
+  const whenFailed = passedExam ? "" : "You did not pass the exam. "
+  if (course.has_to_pay) {
+    return `${whenFailed}If you want to re-take the exam, you need to pay again.`
+  }
+  if (course.can_schedule_exam) {
+    return (
+      <span>
+        {`${whenFailed}You are authorized to take the virtual proctored exam for this course. Please `}
+        <a href="http://edx.org">
```
Comment (noisecapella) [inline static/js/components/dashboard/courses/StatusMessages.js:148]: Can you change this to `https://edx.org`?

**Review by noisecapella — CHANGES_REQUESTED** [review]

File: static/js/components/dashboard/FinalExamCard.js (hunk 3 above)
Comment (noisecapella) [reply static/js/components/dashboard/FinalExamCard.js:271]: No, I'm fine leaving it for now

**Review by noisecapella — APPROVED** [review]
